Getting TypeError when running compile with Docker

Trying the getting started command line. Using the latest docker container. Getting a type error.

livingroom.yaml:
```
esphomeyaml:
  name: livingroom
  platform: ESP8266
  board: d1_mini

wifi:
  ssid: 'Wifi'
  password: 'PASSWORD42'

mqtt:
  broker: '192.168.178.84'
  username: ''
  password: ''

# Enable logging
logger:

ota:
```

Error:
```
$ docker run --rm -v "`pwd`":/config -it ottowinter/esphomeyaml livingroom.yaml compile
INFO [esphomeyaml.config] Reading configuration...
INFO [esphomeyaml.__main__] Generating C++ source...
INFO [esphomeyaml.__main__] Compiling app...
INFO [esphomeyaml.__main__] Running:  platformio run -d livingroom
[Tue Sep 25 22:17:35 2018] Processing livingroom (platform: espressif8266@>=1.8.0; board: d1_mini; framework: arduino)
--------------------------------------------------------------------------------------------------------------------------------------------------------------
LibraryManager: Installing esphomelib
git version 2.11.0
Cloning into '/config/livingroom/.piolibdeps/_tmp_installing-8pKldc-package'...
remote: Enumerating objects: 285, done.
remote: Counting objects: 100% (285/285), done.
remote: Compressing objects: 100% (275/275), done.
remote: Total 285 (delta 35), reused 92 (delta 8), pack-reused 0
Receiving objects: 100% (285/285), 320.49 KiB | 0 bytes/s, done.
Resolving deltas: 100% (35/35), done.
Note: checking out '1fb1a2e2af56625f0821a116720d36e71f4aa365'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

LibraryManager: Installing id=346 @ 0.8.2
Downloading...
Unpacking  [####################################]  100%
LibraryManager: Installing id=305 @ ^1.1.0
Downloading...
Unpacking  [####################################]  100%
LibraryManager: Installing id=1826 @ ^1.0.0
Downloading...
Unpacking  [####################################]  100%
LibraryManager: Installing id=3837 @ 5.13.3
Downloading...
Unpacking  [####################################]  100%
LibraryManager: Installing id=306 @ 1.1.1
Downloading...
Unpacking  [####################################]  100%
LibraryManager: Installing FastLED
git version 2.11.0
Cloning into '/config/livingroom/.piolibdeps/_tmp_installing-NJEDdI-package'...
remote: Enumerating objects: 66, done.
remote: Counting objects: 100% (66/66), done.
remote: Compressing objects: 100% (50/50), done.
remote: Total 10772 (delta 24), reused 34 (delta 16), pack-reused 10706
Receiving objects: 100% (10772/10772), 12.09 MiB | 10.86 MiB/s, done.
Resolving deltas: 100% (7526/7526), done.
HEAD is now at d27bac9 add support for Hallowing! Its a samd21 but pins are rearranged quite a bit (#650)
Verbose mode can be enabled via `-v, --verbose` option
TypeError: __init__() takes exactly 2 arguments (1 given):
File "/usr/local/lib/python2.7/site-packages/platformio/builder/main.py", line 162:
env.SConscript("$BUILD_SCRIPT")
File "/root/.platformio/packages/tool-scons/script/../engine/SCons/Script/SConscript.py", line 541:
return _SConscript(self.fs, *files, **subst_kw)
File "/root/.platformio/packages/tool-scons/script/../engine/SCons/Script/SConscript.py", line 250:
exec _file_ in call_stack[-1].globals
File "/root/.platformio/platforms/espressif8266/builder/main.py", line 49:
@util.memoized()
================================================================= [ERROR] Took 15.00 seconds =================================================================
```

Hello,

Same here, since latest/1.8.0/dev version from yesterday 

`INFO [esphomeyaml.__main__] Running:  platformio run -d /config/volets_velux
[Wed Sep 26 16:43:52 2018] Processing volets_velux (platform: espressif8266@>=1.8.0; board: d1_mini; framework: arduino)
--------------------------------------------------------------------------------
Verbose mode can be enabled via `-v, --verbose` option
TypeError: __init__() takes exactly 2 arguments (1 given):
File "/usr/local/lib/python2.7/site-packages/platformio/builder/main.py", line 162:
env.SConscript("$BUILD_SCRIPT")
File "/root/.platformio/packages/tool-scons/script/../engine/SCons/Script/SConscript.py", line 541:
return _SConscript(self.fs, *files, **subst_kw)
File "/root/.platformio/packages/tool-scons/script/../engine/SCons/Script/SConscript.py", line 250:
exec _file_ in call_stack[-1].globals
File "/root/.platformio/platforms/espressif8266/builder/main.py", line 49:
@util.memoized()
========================== [ERROR] Took 2.34 seconds ==========================`

Revert back to release-1.7.0 docker image: no error

@mehuman , temporary fix: downgrade your docker image, by adding the tag to your command:
`docker run --rm -v "`pwd`":/config -it ottowinter/esphomeyaml:release-v1.7.0 livingroom.yaml compile`

--- a/.travis.yml
+++ b/.travis.yml
@@ -4,7 +4,7 @@ python:
   - "2.7"
 install:
   - pip install -r requirements.txt
-  - pip install tornado esptool flake8==3.5.0 pylint==1.8.4 tzlocal pillow
+  - pip install tornado esptool flake8==3.5.0 pylint==1.9.3 tzlocal pillow
 script:
  - flake8 esphomeyaml
  - pylint esphomeyaml

--- a/.travis.yml
+++ b/.travis.yml
@@ -4,7 +4,7 @@ python:
   - "2.7"
 install:
   - pip install -r requirements.txt
-  - pip install tornado esptool flake8==3.5.0 pylint==1.9.3 tzlocal pillow
+  - pip install flake8==3.5.0 pylint==1.9.3 tzlocal pillow
 script:
  - flake8 esphomeyaml
  - pylint esphomeyaml

--- a/Dockerfile
+++ b/Dockerfile
@@ -10,17 +10,13 @@ EXPOSE 6123
 VOLUME /config
 WORKDIR /usr/src/app
 
-COPY requirements.txt /usr/src/app/
-RUN pip install --no-cache-dir -r requirements.txt && \
-    pip install --no-cache-dir tornado esptool
-
 COPY docker/platformio.ini /usr/src/app/
 RUN platformio settings set enable_telemetry No && \
     platformio run -e espressif32 -e espressif8266; exit 0
 
 COPY . .
 RUN pip install --no-cache-dir -e . && \
-    pip install --no-cache-dir tzlocal
+    pip install --no-cache-dir tzlocal pillow
 
 WORKDIR /config
 ENTRYPOINT ["esphomeyaml"]

--- a/docker/Dockerfile.lint
+++ b/docker/Dockerfile.lint
@@ -3,4 +3,4 @@ FROM python:2.7
 COPY requirements.txt /requirements.txt
 
 RUN pip install -r /requirements.txt && \
-    pip install flake8==3.5.0 pylint==1.8.4 tzlocal pillow
+    pip install flake8==3.5.0 pylint==1.9.3 tzlocal pillow

--- a/requirements.txt
+++ b/requirements.txt
@@ -1,7 +1,7 @@
-voluptuous==0.11.1
-platformio==3.5.2
-pyyaml==3.12
-paho-mqtt==1.3.1
-colorlog==3.1.2
-tornado==5.0.2
-esptool==2.3.1
+voluptuous>=0.11.1
+platformio>=3.5.3
+pyyaml>=3.12
+paho-mqtt>=1.3.1
+colorlog>=3.1.2
+tornado>=5.0.0
+esptool>=2.3.1

--- a/setup.py
+++ b/setup.py
@@ -23,7 +23,7 @@ DOWNLOAD_URL = '{}/archive/{}.zip'.format(GITHUB_URL, const.__version__)
 
 REQUIRES = [
     'voluptuous>=0.11.1',
-    'platformio>=3.5.2',
+    'platformio>=3.5.3',
     'pyyaml>=3.12',
     'paho-mqtt>=1.3.1',
     'colorlog>=3.1.2',

